[13.0][MIG] website

Yasaie opened the pull request:
Migration of `website`
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
@Tecnativa TT23136

Yasaie committed:
[MIG] website

gdgellatly commented on addons/website/migrations/13.0.1.0/pre-migration.py:
```diff
@@ -0,0 +1,27 @@
+def migrate(env, version):
+    openupgrade.rename_models(env.cr, _model_renames)
+    openupgrade.rename_tables(env.cr, _table_renames)
+    openupgrade.rename_fields(env.cr, _field_renames)
```
Get this error on migration

```
Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/service/server.py", line 1179, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/opt/odoo/custom/src/odoo/odoo/modules/registry.py", line 86, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 479, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check, upg_registry)
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 363, in load_marked_modules
    upg_registry=upg_registry,
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 190, in load_module_graph
    migrations.migrate_module(package, 'pre')
  File "/opt/odoo/custom/src/odoo/odoo/modules/migration.py", line 185, in migrate_module
    migrate(self.cr, installed_version)
  File "/usr/local/lib/python3.7/site-packages/openupgradelib/openupgrade.py", line 1778, in wrapped_function
    if use_env2 else cr, version)
  File "/opt/odoo/auto/addons/website/migrations/13.0.1.0/pre-migration.py", line 26, in migrate
    openupgrade.rename_fields(env.cr, _field_renames)
  File "/usr/local/lib/python3.7/site-packages/openupgradelib/openupgrade.py", line 595, in rename_fields
    cr = env.cr
  File "/opt/odoo/custom/src/odoo/odoo/sql_db.py", line 163, in wrapper
    return f(self, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/sql_db.py", line 443, in __getattr__
    return getattr(self._obj, name)
AttributeError: 'psycopg2.extensions.cursor' object has no attribute 'cr'
```
I think rename_fields just expects env, not env.cr

MiquelRForgeFlow committed:
[IMP] website migration

Commit:
[FIX] website: migrate logo

In v12, websites used the company logo. In v13, websites have their own logo.

Without this patch, websites get migrated without logo.

@Tecnativa TT23136

pedrobaeza reviewed (dismissed):
This is missing some work on COW views that my colleague @Yajo is doing, so this can't be merged yet.

Commit:
[FIX] website: add COW autoupdate

Makes use of https://github.com/OCA/openupgradelib/pull/220. See description to understand what these new functions do.

@Tecnativa TT23136

yajo commented:
I rebased to see if it fixes the build.

yajo commented:
https://github.com/OCA/openupgradelib/pull/223 is merged, so I rebuilt, let's see now.

MiquelRForgeFlow commented:
Why last commit changes all lines if the only legit change is adding a line?

yajo commented:
It must be something related to line endings or similar. Let me fix it.

yajo commented:
It should be OK now.

MiquelRForgeFlow commented:
good :+1:

MiquelRForgeFlow commented:
could we merge then?

yajo reviewed (approved):
Yes 😊

pedrobaeza commented on addons/website/migrations/13.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,139 @@
+
+---Fields in module 'website'---
+website      / ir.ui.view               / track (boolean)               : NEW hasdefault
+website      / mrp.document             / key (char)                    : previously in module mrp
```
This is for sure a bug in the `openupgrade_records` module, as there's no `mrp.document` here in this module. Can you please check to see the problem and see if this is important.

pedrobaeza commented on addons/website/migrations/13.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,139 @@
+website      / mrp.document             / key (char)                    : previously in module mrp
+website      / mrp.document             / website_id (many2one)         : previously in module mrp
+website      / mrp.document             / website_url (char)            : previously in module mrp
+website      / res.partner              / is_published (boolean)        : previously in module website_partner
```
You should use openupgradelib's `update_module_moved_fields` method here (and in all moved fields). If not, there some garbage and side effect. See https://github.com/OCA/openupgradelib/commit/6a4360b353986cd2e606b7d1dcc2abfa74f683a6 where I have removed the non correct docstring.

StefanRijnhart replied:
I'm discussing this with Pedro here: https://github.com/OCA/openupgradelib/commit/6a4360b353986cd2e606b7d1dcc2abfa74f683a6 but calling this method should not be necessary.

pedrobaeza commented on addons/website/migrations/13.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,139 @@
+website      / res.users                / is_published (boolean)        : previously in module website_partner
+website      / res.users                / website_published (boolean)   : previously in module website_partner
+website      / res.users                / website_url (char)            : previously in module website_partner
+website      / website                  / default_lang_code (char)      : DEL
```
Please don't group all the fields under a generic "Nothing to do". Explain for each "problematic one" why is "Nothing to do". I'm not sure here that this should be nothing to do. How was this used in previous version? And now has is this inferred or done?

MiquelRForgeFlow replied:
The `code` is taken from `default_lang_id`, and if there is none, then it's taken from `url_code`

pedrobaeza commented on addons/website/migrations/13.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,139 @@
+website      / res.users                / website_published (boolean)   : previously in module website_partner
+website      / res.users                / website_url (char)            : previously in module website_partner
+website      / website                  / default_lang_code (char)      : DEL
+website      / website                  / logo (binary)                 : NEW attachment: True, hasdefault
```
This is not nothing to do, and in fact it was already done by my colleague: https://github.com/OCA/OpenUpgrade/pull/2299/files#diff-a6988dc70066d65d8af32f8557718ee383105841ef721ac1a031cd887156aac4R18

pedrobaeza commented on addons/website/migrations/13.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,139 @@
+website      / website.menu             / group_ids (many2many)         : NEW relation: res.groups
+website      / website.menu             / mega_menu_classes (char)      : NEW
+website      / website.menu             / mega_menu_content (html)      : NEW
+website      / website.rewrite          / route_id (many2one)           : NEW relation: website.route
```
Are you sure this is nothing to do? It sound that you have to do something with this coming from previous version.

pedrobaeza commented on addons/website/migrations/13.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,139 @@
+# NOTHING TO DO
+
+website      / website.rewrite          / name (char)                   : NEW required
+# DONE: post-migration: filled name with some heuristic name
```
This should be done in pre-migration pre-creating the column, for avoiding the constraint error.

pedrobaeza reviewed (changes requested):
Tell me if you are not able to do the changes for me to do them.

TT19838

MiquelRForgeFlow commented:
@pedrobaeza all your comments have been attended
